Dev

Catonpancake opened the pull request.

Catonpancake committed:
merge with main

Catonpancake committed:
week2 quest done. dataloader.py made and all functions and class is moved.

Catonpancake committed:
EDA modified

Catonpancake committed:
EDA merged

Catonpancake committed:
utill file added

Catonpancake committed:
EDA modified

Catonpancake committed:
Merge branch 'dev' of https://github.com/Catonpancake/2023winter-seminar into dev

Catonpancake committed:
Week3 Done

1pha commented on dataloader.py:
```diff
@@ -0,0 +1,38 @@
+        self.img = self.data if not self.istrain else self.data.drop("label", axis=1)
+        self.img = [np.reshape(row[1].values,(28,28)) for row in self.img.iterrows()]
+        self.transform = transform
+        # self.target_transform = target_transform
```
안 쓰는 건 그때그떄 날려버리는 게 좋아요

1pha commented on dataloader.py:
```diff
@@ -0,0 +1,38 @@
+    def __len__(self):
+        return len(self.img)
+
+    def __getitem__(self,idx):
```
Space (self, idx)
typehints: self에는 보통 안붙여줘요 (어차피 class라서). idx: int

1pha commented on dataloader.py:
```diff
@@ -0,0 +1,38 @@
+            return image
+
+        # if self.target_transform:
+        #     label = self.target_transform(label)
```
여기도 안쓰면 바로날려주세요

1pha commented on utils.py:
```diff
@@ -0,0 +1,85 @@
+import os
+os.environ['KMP_DUPLICATE_LIB_OK']='True'
```
https://github.com/NVlabs/stylegan3/issues/75
library 충돌이 나타나서 생기는 이슈로 보이는데, conda 가상환경을 하나 더 만들어서 정리하는 게 좋을 것 같아요.

1pha commented on dataloader.py:
```diff
@@ -0,0 +1,38 @@
+import os
+os.environ['KMP_DUPLICATE_LIB_OK']='True'
```
Library import의 경우는 나눠서 적어주는 것이 좋아요. 전에 몇 번 언급했지만 built-in / third-party / 직접개발한 모듈로 나눠서 쓰면 좋습니다.
이게 처음에 힘들면 isort를 사용해도 좋습니다 https://pycqa.github.io/isort/

1pha commented on utils.py:
```diff
@@ -0,0 +1,85 @@
+from PIL import Image
+import matplotlib.pyplot as plt
+
+def reshape(imgdata: pd.DataFrame()):
```
typehint에는 class가 들어가야해요 (imgdata: pd.DataFrame) DataFrame 뒤에 () 지워주세요

Catonpancake replied:
Fixed

1pha replied:
"Resolve conversation" 눌러주시면됩니당

1pha commented on utils.py:
```diff
@@ -0,0 +1,85 @@
+
+def reshape(imgdata: pd.DataFrame()):
+    img_reshaped = np.reshape(imgdata.values,(28,28))
+
```
enter remove

Catonpancake replied:
Fixed

1pha commented on utils.py:
```diff
@@ -0,0 +1,85 @@
+    return img_reshaped
+
+def drawimg(img, label=None):
+    if label == None:
+        plt.title(f'Label: {img[1]}')
+    else:
+        plt.title(f'Label: {label}')
```
if문을 줄여줄 수 있어요
`plt.title(f"Label: {img[1] if label is None else label})`
혹은 더 간단하게
`plt.title(f"Label: {label or img[1]}")`
하지만 후자는 label과 img[1]에 뭐가 들어오느냐에 따라 잘 고려해야해요. or문은 둘 다 None이 아니고 유효한 값이 올 때 뒤의 값을 우선으로 사용합니다. (2 or 1은 1을 return, 1 or 2는 2를 return. 1 or None, None or 1은 모두 1을 반환하구요)

Catonpancake replied:
오 엄청 짧아지는군요

1pha commented on utils.py:
```diff
@@ -0,0 +1,85 @@
+    else:
+        plt.title(f'Label: {label}')
+    plt.imshow(img[0], cmap='gray')
+
```
함수들 사이는 두 번 엔터를 쳐줍니다. 사소하지만 습관들여두면 좋아요!

Catonpancake replied:
Fixed

1pha commented on utils.py:
```diff
@@ -0,0 +1,85 @@
+        ax.imshow(tensor_to_pltimg(data), cmap='gray')
+        
+        
+def train(dataloader, model, loss_fn, optimizer, device):
```
전체적인 맥락에서 `train`과 `test`함수는 다른 스크립트로 분리되어 있는 게 좋을 것 같습니다.

1pha replied:
typehint도 적어주면 좋아요

1pha commented on utils.py:
```diff
@@ -0,0 +1,85 @@
+            correct /= size
+            print(f"Test Error: \n Accuracy: {(100*correct):>0.1f}%, Avg loss: {test_loss:>8f} \n")
+        elif valid == False:
+            with open('Submission.csv', 'a+', newline='') as f:  # 'a+' means append to a file
```
이러면 같은 파일위에서 계속 작업하지 않나요? `test` 함수의 입력에서 fname을 따로 받아주거나, 같은 파일이 존재할 때 처리할 방법을 따로 적어주는 것이 좋을 것 같습니다.

Catonpancake replied:
오 맞아요 이거 해두려고 했는데 고민되는게 버전번호를 넣는게 나을지 아니면 model 명이나 transform 변화같이 바뀌는 parameter들이 바뀌면 자연스럽게 제목이 바뀌게 하는 게 나을 지 고민중이었습니다. 보통은 어떻게 하는지 궁금하네요

1pha reviewed (changes requested):
제가 전에 봤을 때보다 정말 많이 깔끔해지신 것 같아요. 요청드린 사항들 반영해서 commit 넣어주시면 다시 봐드릴게요

Catonpancake committed:
All minor issues are fixed.
1) Environmental issue on torch should be fixed
2) Cuda is not activated due to torch issue
3) FIle saving on test function has been changed. Need test to verify

Catonpancake committed:
Unused part removed. type hint added

Catonpancake committed:
variable not assigned is fixed

Catonpancake committed:
Merge branch 'dev' of https://github.com/Catonpancake/2023winter-seminar into dev

Catonpancake committed:
train test function is divided into different .py file

Catonpancake committed:
Rename trainer file

1pha commented on utils_traintest.py:
```diff
@@ -0,0 +1,59 @@
+from dataloader import DigitDataset
+from utils import reshape, drawimg, tensor_to_pltimg, multidrawimg, train, test
+import torch
+from torch import nn
+from torch.utils.data import Dataset
+from torch.utils.data import DataLoader
+import os
+os.environ['KMP_DUPLICATE_LIB_OK']='True'
+
+import csv
```
```suggestion
import os
import csv
os.environ['KMP_DUPLICATE_LIB_OK']='True'

import torch
from torch import nn
from torch.utils.data import Dataset
from torch.utils.data import DataLoader

from dataloader import DigitDataset
import utils as ut
```

그리고 뒤에 reshape이나 여러 utils에서 온 함수들 ut.reshape으로 사용하기

Catonpancake committed:
Trainer class is ongoing
